app/screens/client_upgrade/client_upgrade.js

```diff
@@ -0,0 +1,322 @@
+        forceUpgrade: PropTypes.bool,
+        intl: intlShape.isRequired,
+        latestVersion: PropTypes.string,
+        navigator: PropTypes.object,
+        theme: PropTypes.object.isRequired,
+        upgradeType: PropTypes.string
+    };
+
+    state = {
+        upgradeType: UpgradeTypes.NO_UPGRADE
+    }
+
+    componentDidMount() {
+        if (this.props.userCheckedForUpgrade) {
+            this.checkUpgrade(this.props);
+        }
+    }
+
+    checkUpgrade = ({minVersion, latestVersion}) => {
+        const {actions, currentVersion, downloadLink} = this.props;
+
+        // We need at least minVersion or latestVersion and the app downloadlink
+        if (!(latestVersion || minVersion) || !downloadLink) {
+            return;
+        }
+
+        const upgradeType = checkUpgradeType(currentVersion, minVersion, latestVersion, actions.logError);
+
+        if (upgradeType === UpgradeTypes.NO_UPGRADE) {
+            return;
+        }
+
+        this.setState({
+            upgradeType
+        });
+
+        actions.setLastUpgradeCheck();
+    }
+
+    handleClose = () => {
+        if (this.props.closeAction) {
+            this.props.closeAction();
+        } else if (this.props.userCheckedForUpgrade) {
+            this.props.navigator.pop();
+        } else {
+            this.props.navigator.dismissModal();
+        }
+    }
+
+    handleDownload = () => {
+        const {downloadLink, intl} = this.props;
+
+        Linking.canOpenURL(downloadLink).then((supported) => {
+            if (supported) {
+                return Linking.openURL(downloadLink);
+            }
+
+            Alert.alert(
+                intl.formatMessage({
+                    id: 'mobile.client_upgrade.download_error.title',
+                    defaultMessage: 'Upgrade Error'
+                }),
+                intl.formatMessage({
+                    id: 'mobile.client_upgrade.download_error.message',
+                    defaultMessage: 'An error occurred while trying to open the download link.'
+                })
+            );
+
+            return false;
+        });
+    }
+
+    handleUpgrade = () => {
+        const {actions, downloadLink} = this.props;
+
+        try {
+            Linking.openURL(downloadLink);
+        } catch (error) {
+            actions.logError(error);
+        }
+    }
+
+    renderMustUpgrade() {
+        const {theme} = this.props;
+        const styles = getStyleFromTheme(theme);
+
+        return (
+            <View style={styles.messageContent}>
+                <FormattedText
+                    id='mobile.client_upgrade.must_upgrade_title'
+                    defaultMessage='Upgrade Required'
+                    style={styles.messageTitle}
+                />
+                <FormattedText
+                    id='mobile.client_upgrade.must_upgrade_subtitle'
+                    defaultMessage={'A client upgrade is required before continuing to use the app.'}
```

Comment:
you don't need to wrap the string in {} here, no need to change but just to make you aware